### PR TITLE
Populate Kafka header with bootc information.

### PR DESCRIPTION
**Pull request by roliveri**
# Overview

This PR is being created to address [RHINENG-9536](https://issues.redhat.com/browse/RHINENG-9536).
(A description of your PR's changes, along with why/context to the PR, goes here.)

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Include raw query examples in the PR description, if adding/modifying SQL query
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by skateman — APPROVED** [review]
The Seal of Approval

**Review by thearifismail — COMMENTED** [review]
supporting @fstavela comment

**Review by kruai — APPROVED** [review]

**Review by kruai — CHANGES_REQUESTED** [review]
Oops - didn't see the IQE test failures and Fero's comment. Implementation looks good, but it sounds like the header needs to be a string.

File: app/queue/events.py [hunk 1]
```diff
@@ -87,6 +92,7 @@ def message_headers(
         "reporter": reporter,
         "host_type": host_type,
         "os_name": os_name,
+        "is_bootc": is_bootc,
```
[the remark this answers is not in the document]
Comment (roliveri) [reply app/queue/events.py:95]: @fstavela 
I'm not seeing this issue in the unit tests. Does that mean the EventProducer isn't properly covered in our unit tests?

[the remark this answers is not in the document]
Comment (roliveri) [reply app/queue/events.py:95]: @fstavela 
I've updated the code, changing `is_bootc` from a boolean to string representation ("True" of "False").

@skateman 
Please note the change in the datatype of `is_bootc` in the event header.

**Comment by roliveri** [comment]
@fstavela 
It looks like most, if not all, of the remaining iqe test failures should be resolved by updating `events_message_expected_headers()`, in `mq_utils.py`, to include the `is_bootc` header.

**Review by fstavela — APPROVED** [review]

**Comment by fstavela** [comment]
> @fstavela It looks like most, if not all, of the remaining iqe test failures should be resolved by updating `events_message_expected_headers()`, in `mq_utils.py`, to include the `is_bootc` header.

You are right, everything looks good now. I have an MR with the necessary updates ready, but I can't merge it until this PR is merged, because then the tests would be failing on other PRs. If it's possible, I suggest merging this PR even with the failed PR checks and then I will update the tests immediately. If it's not possible, I can skip the tests first, let the PR check pass and then I can update and reenable them.

**Review by thearifismail — APPROVED** [review]

**Pull request merged**